### ISSUE TITLE
[SfRadialGauge] cornerStyle not working for low values

### ISSUE THREAD

**Issue by stefanoromanello** [syncfusion/flutter-widgets#99]
I'm facing the following issue. When low values are assing to the gauge the left corner is flat instead of round. 

```
SfRadialGauge(axes: <RadialAxis>[
                    RadialAxis(
                      minimum: 0,
                      maximum: 100,
                      showLabels: false,
                      showTicks: false,
                      annotations: <GaugeAnnotation>[
                        GaugeAnnotation(
                          positionFactor: 0.1,
                          angle: 90,
                          widget: Text(
                            (percentage * 100).toStringAsFixed(0) + ' / 100',
                            textAlign: TextAlign.center,
                            style: TextStyle(
                              fontSize: ScreenUtil().setSp(15),
                            ),
                          ),
                        ),
                      ],
                      pointers: <GaugePointer>[
                        RangePointer(
                          value: percentage * 100,
                          cornerStyle: CornerStyle.bothCurve,
                          animationDuration: 500,
                          width: 0.2,
                          color: MyTheme.primaryColor,
                          sizeUnit: GaugeSizeUnit.factor,
                        )
                      ],
                      axisLineStyle: AxisLineStyle(
                        thickness: 0.2,
                        cornerStyle: CornerStyle.bothCurve,
                        color: MyTheme.cardColor(context),
                        thicknessUnit: GaugeSizeUnit.factor,
                      ),
                    )
                  ]),
```

WIth the same code but different values I get:

![image](https://user-images.githubusercontent.com/4083715/103172927-7b204880-4857-11eb-9207-47121aac7c88.png)

![image](https://user-images.githubusercontent.com/4083715/103172935-8ecbaf00-4857-11eb-88db-d188668230d1.png)

**Comment by Dharanipalani** [comment]
Hi @stefanoromanello, 

  Greetings from Syncfusion. We analyzed your query at our end and confirmed that is the behavior when we set a rounded corner for the start position of the range pointer. When the pointer value is below the interval value which is calculated internally (maximum labels as 3 for 100 pixels as like in the chart), the rounded corner will change into a flat in the start position alone.


Thanks,
Dharanitharan.P

**Comment by stefanoromanello** [comment]
> Hi @stefanoromanello,
> 
> Greetings from Syncfusion. We analyzed your query at our end and confirmed that is the behavior when we set a rounded corner for the start position of the range pointer. When the pointer value is below the interval value which is calculated internally (maximum labels as 3 for 100 pixels as like in the chart), the rounded corner will change into a flat in the start position alone.
> 
> Thanks,
> Dharanitharan.P

Thanks for the response. Is there a way to avoid this? I've understood that you're doing this because if a value is below some threshold both the corners overlap but I'm using multiples gauges and seeing someone with round corners and someone with flat ones are not great. 
Is it possible for example to display a simple dot (both round corners near each other with the flat side attached) that represents all the values under your interval instead of changing the style to the flat one?
If the answer is yes can you please tell me what to change in your library to archive this?

Thanks

**Comment by SriramKiranSenthilkumar** [comment]
Hi @stefanoromanello,

Thanks for the revert. We have analyzed your query at our end and as per your request, we have created a simple radial gauge sample in which we have used Marker pointers along with range pointer to achieve the requested scenario regarding rendering the axis line and range pointer in rounded corners even when the value goes below the threshold. We have also attached the sample below in which we also have explained the workaround process in detail. Kindly please check the below-attached sample and revert us if you still have further concerns on this.
**Screenshot:**
![image](https://user-images.githubusercontent.com/63577117/103299158-e9186b80-4a21-11eb-81b2-a36d930eff8f.png)

Sample:  [sample.zip](https://github.com/syncfusion/flutter-widgets/files/5751395/sample.zip)

Regards,
Sriram Kiran

**Comment by stefanoromanello** [comment]
Nice, thanks!!

**Comment by berkaayildiz** [comment]
I don't believe that adding a circle to the beginning of the gauge is a viable solution. This causes the gauge's left side to be larger and longer than the right side, resulting in an asymmetric design that is not preferred.  Because of the size of the circle, the gauge will not reflect any progress for minor progresses using this technique. Aside from that, this would add unnecessary boilerplate logic for color selection, such as when the gauge is at zero, the circle should be grey, and this won't even work for gradient colors.

I think this should be done with clipping the starting point, or adding an option to do that.

**Comment by berkaayildiz** [comment]
I don't believe adding a circle to the beginning of the gauge is a viable solution. This causes the gauge's left side to be larger and longer than the right side, resulting in an asymmetric design that is not preferred.  Because of the added circle, the gauge will not reflect any progress for minor progresses using this technique. Aside from that, this would add unnecessary boilerplate logic for color selection, such as when the gauge is at zero, the circle should be set back to grey, and maintaining the color correctness won't even work for gradient colors.

I think this should be fixed with clipping the starting point, or adding an option to do that. I think this issue should be re-opened.

**Comment by HashRay** [comment]
thanku for the fix :)

**Comment by vasanth5545** [comment]
The named parameter 'cornerStyle' isn't defined.

Problem how to fix?

**Comment by Mugunthan-Ramalingam** [comment]
Hi @vasanth5545,

We have checked the mentioned issue and tried to replicate it in [SfRadialGauge](https://pub.dev/documentation/syncfusion_flutter_gauges/latest/gauges/SfRadialGauge-class.html) with version [29.2.10](https://pub.dev/packages/syncfusion_flutter_charts/versions/29.2.10) by considering the following scenario,

1. Ensured the simple sample in web, windows and android platform.
2.  Ensured the sample by setting the [cornerStyle](https://pub.dev/documentation/syncfusion_flutter_gauges/latest/gauges/AxisLineStyle/cornerStyle.html) property.

	
However, we were unable to reproduce it on our end. Please check the attached sample, and if you are still experiencing the issue, we request that you replicate it in the attached sample and provide us with more details regarding the specific scenario in which you are encountering this issue. This will help us to assist you more effectively.
 
 Demo:
 
![Image](https://github.com/user-attachments/assets/87815e08-5094-4bb0-abb5-f3c9d71c4a63)

Sample: [gh_99.zip](https://github.com/user-attachments/files/20775035/gh_99.zip)


Regards,
Mugunthan.

**Comment by vasanth5545** [comment]
This widget not working sir ... I was changed custompainter widget... still working sir ... But I think 🤔 why not working sfRadialgauge?  ...

**Comment by ManiKesavaraj** [comment]
Hi @vasanth5545 ,

We are unable to clearly understand your issue based on the details you provided. Therefore, to assist you better, please provide the following information to proceed further:

- Specific product version and Flutter SDK version in which the issue occurs.
- In which platform does the mentioned issue occurs? (e.g., iOS, Android, Web etc.).
- Replication sample to reproduce the mentioned issue.
- Screenshots or a demo that replicates the issue.

These details will help us investigate further and provide an appropriate solution.

Regards,
Naveen K.

**Comment by AliHaxxan2004** [comment]
@ManiKesavaraj the issue appears when we increase the width of RangePointer above 10 while the value is under 5 or 10. like this....
`RangePointer(value: 3, cornerStyle: CornerStyle.bothCurve, width: 20, ),`

**Comment by Mugunthan-Ramalingam** [comment]
Hi @AliHaxxan2004,


In [SfRadialGauge](https://pub.dev/documentation/syncfusion_flutter_gauges/latest/gauges/SfRadialGauge-class.html), the CornerStyle.bothCurve option does not render the [RangePointer](https://pub.dev/documentation/syncfusion_flutter_gauges/latest/gauges/RangePointer-class.html) corner curves properly when the pointer value is too low (approximately less than 15% of the total range). This is because, there isn't enough space to display the curved corners. This is the current behavior of the [RangePointer](https://pub.dev/documentation/syncfusion_flutter_gauges/latest/gauges/RangePointer-class.html) when using the [CornerStyle](https://pub.dev/documentation/syncfusion_flutter_gauges/latest/gauges/RangePointer/cornerStyle.html) property.
 
However, we have already logged a feature request to enhance the corner style logic for low values in our Feedback Portal and this feature will be available in any of our upcoming release. You can track the status of this request through the provided feedback link.
 
FR Link: [Improve the range point corner rendering when the values are low in Flutter | Feedback Portal](https://www.syncfusion.com/feedback/31915/improve-the-range-point-corner-rendering-when-the-values-are-low)


Additionally, to render rounded corners for low values, we’ve prepared a simple sample that combines [MarkerPointer](https://pub.dev/documentation/syncfusion_flutter_gauges/latest/gauges/MarkerPointer-class.html) at both ends of the range along with the [RangePointer](https://pub.dev/documentation/syncfusion_flutter_gauges/latest/gauges/RangePointer-class.html). The circular markers placed at the start and end will simulate curved corners visually, enhancing the appearance even when the pointer value is low. We’ve shared the sample for your reference, and you can customize it further based on your requirements.

![Image](https://github.com/user-attachments/assets/06a874e3-4740-4e9f-8984-b00f9159afba)

Sample: [gh_99.zip](https://github.com/user-attachments/files/22250575/gh_99.zip)

 
Regards,
Mugunthan.
